.github/workflows/release.yml

```diff
@@ -0,0 +1,162 @@
+name: Release
+
+on:
+  pull_request:
+    branches:
+      - '*'  # must quote since "*" is a YAML reserved character; we want a string
+    paths-ignore:
+      - '.github/workflows/quarto-render.yml'
+      - '_quarto.yml'
+      - 'quarto-materials/*'
+      - '**/.md'
+      - 'tiledb/doxygen/source/*'
+      - 'tiledb/sm/c_api/tiledb_version.h'
+  push:
+    branches:
+      - dev
+      - 'release-*'
+    tags:
+      - '*'
+
+jobs:
+  Build-Release:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [macos-arm64, macos-x86_64, linux-x86_64, linux-x86_64-noavx2, windows-x86_64]
+        include:
+          - platform: windows-x86_64
+            os: windows-latest
+            triplet: x64-windows-release
+          - platform: linux-x86_64
+            os: ubuntu-latest
+            manylinux: true
+            triplet: x64-linux-release
+          - platform: linux-x86_64-noavx2
+            os: ubuntu-latest
+            cmake_args: -DCOMPILER_SUPPORTS_AVX2=OFF
+            triplet: x64-linux-release
+            manylinux: true
+          - platform: macos-x86_64
+            os: macos-latest
+            triplet: x64-osx-release
+          - platform: macos-arm64
+            os: macos-latest
+            cmake_args: -DCMAKE_OSX_ARCHITECTURES=arm64
+            triplet: arm64-osx-release
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.manylinux && 'quay.io/pypa/manylinux2014_x86_64' || '' }}
```

Comment:
I would lean toward pinning the image version in order to avoid random breakage when it is changed (which has happened in the past).

Dependabot [might (?)](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file?learn=dependency_version_updates&learnProduct=code-security#docker) be able to update the image on a schedule, which would let us ensure smooth changes (not necessary in this PR though).

Reply:
Pinned. I think the Docker section of the Dependabot docs refers to images in Dockerfiles. I don't think these containers can be automatically updated; they are random strings in the workflow file.